src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -1873,25 +1873,24 @@ bool DoorLockServer::findUserIndexByCredential(chip::EndpointId endpointId, Cred
     return false;
 }
 
-Status DoorLockServer::createUser(chip::EndpointId endpointId, chip::FabricIndex creatorFabricIdx, chip::NodeId sourceNodeId,
-                                  uint16_t userIndex, const Nullable<chip::CharSpan> & userName,
-                                  const Nullable<uint32_t> & userUniqueId, const Nullable<UserStatusEnum> & userStatus,
-                                  const Nullable<UserTypeEnum> & userType, const Nullable<CredentialRuleEnum> & credentialRule,
-                                  const Nullable<CredentialStruct> & credential)
+StatusIB DoorLockServer::createUser(chip::EndpointId endpointId, chip::FabricIndex creatorFabricIdx, chip::NodeId sourceNodeId,
+                                    uint16_t userIndex, const Nullable<chip::CharSpan> & userName,
+                                    const Nullable<uint32_t> & userUniqueId, const Nullable<UserStatusEnum> & userStatus,
+                                    const Nullable<UserTypeEnum> & userType, const Nullable<CredentialRuleEnum> & credentialRule,
+                                    const Nullable<CredentialStruct> & credential)
 {
     EmberAfPluginDoorLockUserInfo user;
     if (!emberAfPluginDoorLockGetUser(endpointId, userIndex, user))
     {
         ChipLogError(Zcl, "[createUser] Unable to get the user from app [endpointId=%d,userIndex=%d]", endpointId, userIndex);
-        return Status::Failure;
+        return StatusIB(Status::Failure);
     }
 
     // appclusters, 5.2.4.34: to modify user its status should be set to Available. If it is we should return OCCUPIED.
     if (UserStatusEnum::kAvailable != user.userStatus)
     {
         ChipLogProgress(Zcl, "[createUser] Unable to overwrite existing user [endpointId=%d,userIndex=%d]", endpointId, userIndex);
-        // TODO FixMe  https://github.com/project-chip/connectedhomeip/issues/32100
-        return static_cast<Status>(DlStatus::kOccupied);
+        return StatusIB(Status::Failure, to_underlying(DlStatus::kOccupied));
```

Comment:
What is the difference between using `StatusIB` and `ClusterStatusCode`?

Reply:
That is a good question, `ClusterStatusCode` doesn't seems to be used in the SDK at all and `StatusIB` was actually recommended by @bzbarsky-apple.... Any thoughts on that ?

Reply:
Use `ClusterStatusCode` instead

Reply:
done